packages/core/admin/_internal/node/createBuildContext.ts

```diff
@@ -170,11 +170,14 @@ const createBuildContext = async ({
 
   const features = strapiInstance.config.get('features', undefined);
 
+  const serveAdminPanel = strapiInstance.config.get('admin.serveAdminPanel', true);
+
   const { bundler = 'webpack', ...restOptions } = options;
 
   const buildContext = {
     appDir,
-    basePath: `${adminPath}/`,
+    // If serveAdminPanel is false, use relative paths for separate deployment
+    basePath: serveAdminPanel ? `${adminPath}/` : strapiInstance.config.get('admin.url', ''),
```

Comment:
```suggestion
    basePath: serveAdminPanel ? `${adminPath}/` : strapiInstance.config.get('admin.url', `${adminPath}/`),
```
I'm concerned this will be a breaking change for someone if we change the default values. This will allow it to continue working exactly as before if admin.url isn't set, but now fixes the code to match the documentation, where it says to use admin.url to set this value.